### PR TITLE
Update metrics collector to the released image

### PR DESCRIPTION
**Pull request by cristipogacean**
Update metrics collector to the released image

### CODE REVIEW

**Comment by prmathur-microsoft** [comment]
Two Questions - 
1. Is the discovery module Fix approved for LTS?
2. Is the metrics collector fix approved for LTS?

If so, can it be split into separate commits (for easier revert if needed)

**Review by koepalex — APPROVED** [review]
:shipit:

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Registry/src/Deploy/IoTHubDiscovererDeployment.cs [hunk 1]
```diff
@@ -95,15 +95,7 @@ public sealed class IoTHubDiscovererDeployment : IHostProcess {
                     Cmd = new[] {
                         "PkiRootPath=/mount/pki"
                     },
-                    NetworkingConfig = new {
```
Comment (marcschier) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Registry/src/Deploy/IoTHubDiscovererDeployment.cs:98]: Can we break this file into a separate PR?  is the discovery module still able to enumerate the subnets of the network adapters?  Or does this require now to also specify subnet mask?

File: common/src/Microsoft.Azure.IIoT.Core/src/Deploy/ILogWorkspaceConfig.cs [hunk 2]
```diff
@@ -19,5 +19,10 @@ public interface ILogWorkspaceConfig {
         /// Log Analytics Workspace Key
         /// </summary>
         string LogWorkspaceKey { get; }
+
+        /// <summary>
+        /// The Resource Is of the IoT Hub
```
Comment (marcschier) [inline common/src/Microsoft.Azure.IIoT.Core/src/Deploy/ILogWorkspaceConfig.cs:24]: typo

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Registry/src/Deploy/IoTHubMetricsCollectorDeployment.cs [hunk 3]
```diff
@@ -108,11 +106,20 @@ public sealed class IoTHubMetricsCollectorDeployment : IHostProcess {
...
                                 ""value"": """ + _config.LogWorkspaceId + @"""
                             },
-                            ""AzMonWorkspaceKey"": {
+                            ""LogAnalyticsSharedKey"": {
                                 ""value"": """ + _config.LogWorkspaceKey + @"""
+                            },
+                            ""ResourceId"": {
+                                ""value"": """ + _config.IoTHubResourceId + @"""
+                            },
+                            ""MetricsEndpointsCSV"": {
```
Comment (marcschier) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Registry/src/Deploy/IoTHubMetricsCollectorDeployment.cs:121]: Should we add discovery module as well?  Does this value support wildcards?

**Review by marcschier — CHANGES_REQUESTED** [review]

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Registry/src/Deploy/IoTHubDiscovererDeployment.cs (hunk 1 above)
Comment (cristipogacean) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Registry/src/Deploy/IoTHubDiscovererDeployment.cs:98]: sure, I will move this to a separate PR. 
with this change the user needs to specify the subnet mask, in a same manner as it needs to be done for Windows containers. Do you think this is a blocker?

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Registry/src/Deploy/IoTHubMetricsCollectorDeployment.cs (hunk 3 above)
Comment (cristipogacean) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Registry/src/Deploy/IoTHubMetricsCollectorDeployment.cs:121]: For some reason the discovery was not part of this in the first place. That's the only reason I did not add it. 
I did not find any reference in the documentation on supporting wildcards.

**Review by marcschier — APPROVED** [review]

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Registry/src/Deploy/IoTHubDiscovererDeployment.cs (hunk 1 above)
Comment (marcschier) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Registry/src/Deploy/IoTHubDiscovererDeployment.cs]: Yes, let's understand the root cause a bit more before we make the other change.  We can deal with it in the other PR.
